src/redux/reducers/todosReducer.js

```diff
@@ -1,13 +1,32 @@
+import { ADD_TODO, IS_ADDING } from "redux/actions/actionTypes"
+import { formDate } from "utils/formDate"
 export const initialState = {
-  todosItems: [
-    { text: "Eat", completed: true },
-    { text: "Sleep", completed: false },
-  ],
+  todoItems: [],
   visibilityFiler: "SHOW_ALL",
+  isAdding: false,
 }
 
 export default function todosReducer(state = initialState, action) {
   switch (action.type) {
+    case ADD_TODO:
+      const { text } = action.payload
+      return {
+        ...state,
+        todoItems: [
+          ...state.todoItems,
+          {
+            text: text,
+            createdAt: formDate(new Date()),
```

Comment:
Reducers are what called pure functions. They must be deterministic in nature, meaning for each set...in nature such as calling API, random functions, date functions inside reducers.
More [here](https://redux.js.org/style-guide/#reducers-must-not-have-side-effects)